internal/reflect/into.go

```diff
@@ -109,11 +111,19 @@ func BuildValue(ctx context.Context, typ attr.Type, val tftypes.Value, target re
 		// all that's left to us now is to set it as an empty value or
 		// throw an error, depending on what's in opts
 		if !opts.UnhandledUnknownAsEmpty {
-			err := fmt.Errorf("unhandled unknown value")
+			typTypeStr := reflect.TypeOf(typ).String()
+
+			if typTypeStr == "types.primitive" {
+				typTypeStr = typ.String()
+			}
+
+			typTypeStr = strings.TrimSuffix(typTypeStr, "Type")
+
 			diags.AddAttributeError(
 				path,
 				"Value Conversion Error",
-				"An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider developer:\n\n"+err.Error(),
+				"An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider developer:\n\n"+
+					fmt.Sprintf("Received unknown value for %s, however the current struct field type %s cannot handle unknown values. Use %s, or a custom type that supports unknown values instead.", path, target.Type(), typTypeStr),
```

Comment:
Given that the target type strings can be quite lengthy, I wonder if we should use additional newlines in the diagnostics to potentially make things more readable?

```
Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or a custom type that handles unknown values.

Path: path.String()
Target Type: target.Type()
Suggested Type: typ.String()
```

Reply:
By the way, thank you for adding the path information here. Since it is probably not obvious at all, we should prefer to include path information in the diagnostic because the Terraform user interface does not output anything regarding the path if there's no associated configuration of the path value. So this covers cases like `Computed` only attribute paths. Terraform also doesn't support diagnostic paths for set elements across the protocol, so this also covers cases with set-based schemas.

Reference: https://github.com/hashicorp/terraform/issues/31575
Reference: https://github.com/hashicorp/terraform/blob/4153685bfe0c471a935f26a657d4351fb927313a/docs/plugin-protocol/tfplugin6.3.proto#L43-L56

Reply:
I've restructured the diagnostics as you suggested for `unknown` and have done something similar for `null`:

```
Received null value, however the target type cannot handle null values. Use the corresponding `types` package type, a pointer type or a custom type that handles null values.

Path: path.String()
Target Type: target.Type()
Suggested `types` Type: typ.String()
Suggested Pointer Type: *target.Type()
```

Reply:
Loving it